Formula/jenv.rb

```diff
@@ -15,12 +15,23 @@ def install
   end
 
   def caveats
-    <<~EOS
-      To activate jenv, add the following to your #{shell_profile}:
+    if preferred == :fish
+      <<~EOS
+        To activate jenv, run the following commands:
 
-        export PATH="$HOME/.jenv/bin:$PATH"
-        eval "$(jenv init -)"
-    EOS
+          echo 'status --is-interactive; and source (jenv init -|psub)' >> #{shell_profile}
+          set -q XDG_CONFIG_HOME; or set XDG_CONFIG_HOME $HOME
+          ln -sf #{opt_libexec}/fish/*.fish $XDG_CONFIG_HOME/.config/fish/functions
```

Comment:
If we need to symlink the `.fish` files into `~/.config/fish/functions`, wouldn't it be better to install them into the right location instead? i.e. using the variable [`fish_function`](https://rubydoc.brew.sh/Formula.html#fish_function-instance_method). For example, with something like:
```ruby
fish_function.install Dir[libexec/"fish/*.fish"]
```
in the `install` method.

Reply:
SGTM, feel free to update this branch with that change or suggest one here I can commit. I'm just not sure the exact code you had in mind here.

Reply:
Something like this:
```diff
diff --git a/Formula/jenv.rb b/Formula/jenv.rb
index a1f3a7e649..7874997525 100644
--- a/Formula/jenv.rb
+++ b/Formula/jenv.rb
@@ -12,6 +12,7 @@ class Jenv < Formula
   def install
     libexec.install Dir["*"]
     bin.write_exec_script libexec/"bin/jenv"
+    fish_function.install Dir[libexec/"fish/*.fish"]
   end
 
   def caveats
```
I'd want to test this doesn't break anything first, since `#install` will move the files from `libexec/"fish"` into the `fish_function` directory (`share/"fish/vendor_functions.d"`).

If it turns out moving those files breaks things, we can try `#install_symlink` instead of `#install`.

Also, assuming that one of these is the right thing to do, presumably the caveats need to be adjusted (as presumably, we would no longer need symlinks into `~/.config`).

Reply:
`#install_symlink` would make homebrew installs more like non-homebrew fish installs, which might be desirable. But it seems like either _should_ work.

Reply:
Maybe `#install_symlink` is better then. `#install` could work now, but I don't think we can rule out it causing problems later on.

Reply:
Neither `#install_symlink` nor `#install` seem to work. Any ideas?

```
brew uninstall --force jenv                                                                                                                            ⮂ 13:37:33 ⮂ 2021-06-02
Uninstalling jenv... (84 files, 73.1KB)
 dxia@dxia-mbp ⮀ /o/h/L/T/h/homebrew-core ⮀ ⭠ jenv-caveat-fish  ⮀
 ⮀ la ~/.config/fish/functions/                                                                                                                           ⮂ 13:37:44 ⮂ 2021-06-02
total 32
drwxr-xr-x   8 dxia  staff   256B May 12 07:01 .
drwx------  16 dxia  staff   512B May 12 07:01 ..
-rw-r--r--   1 dxia  staff   816B Feb 17  2019 burk.fish
lrwxr-xr-x   1 dxia  staff    47B May  8 13:47 export.fish -> /opt/homebrew/opt/jenv/libexec/fish/export.fish
-rw-r--r--   1 dxia  staff   5.2K Sep  9  2014 fish_prompt.fish
-rw-r--r--   1 dxia  staff    55B Feb 13  2019 fish_user_key_bindings.fish
lrwxr-xr-x   1 dxia  staff    45B Apr  8 15:11 fzf_key_bindings.fish -> /opt/homebrew/opt/fzf/shell/key-bindings.fish
lrwxr-xr-x   1 dxia  staff    45B May  8 13:47 jenv.fish -> /opt/homebrew/opt/jenv/libexec/fish/jenv.fish
 dxia@dxia-mbp ⮀ /o/h/L/T/h/homebrew-core ⮀ ⭠ jenv-caveat-fish  ⮀
 ⮀ rm ~/.config/fish/functions/jenv.fish ~/.config/fish/functions/export.fish                                                                             ⮂ 13:37:46 ⮂ 2021-06-02
 dxia@dxia-mbp ⮀ /o/h/L/T/h/homebrew-core ⮀ ⭠ jenv-caveat-fish  ⮀
 ⮀ la ~/.config/fish/functions/                                                                                                                           ⮂ 13:38:05 ⮂ 2021-06-02
total 32
drwxr-xr-x   6 dxia  staff   192B Jun  2 13:38 .
drwx------  16 dxia  staff   512B May 12 07:01 ..
-rw-r--r--   1 dxia  staff   816B Feb 17  2019 burk.fish
-rw-r--r--   1 dxia  staff   5.2K Sep  9  2014 fish_prompt.fish
-rw-r--r--   1 dxia  staff    55B Feb 13  2019 fish_user_key_bindings.fish
lrwxr-xr-x   1 dxia  staff    45B Apr  8 15:11 fzf_key_bindings.fish -> /opt/homebrew/opt/fzf/shell/key-bindings.fish
 dxia@dxia-mbp ⮀ /o/h/L/T/h/homebrew-core ⮀ ⭠ jenv-caveat-fish  ⮀
 ⮀ brew install --build-from-source jenv                                                                                                                  ⮂ 13:38:06 ⮂ 2021-06-02
==> Downloading https://github.com/jenv/jenv/archive/0.5.4.tar.gz
Already downloaded: /Users/dxia/Library/Caches/Homebrew/downloads/f9afe32d5657c7d7caf6d54af2c81e10659e0f46696133296bbc3cb6cc20cead--jenv-0.5.4.tar.gz
==> Caveats
To activate jenv, run the following commands:

  echo 'status --is-interactive; and source (jenv init -|psub)' >> ~/.config/fish/config.fish


fish functions have been installed to:
  /opt/homebrew/share/fish/vendor_functions.d
==> Summary
🍺  /opt/homebrew/Cellar/jenv/0.5.4: 84 files, 73KB, built in 2 seconds
 dxia@dxia-mbp ⮀ /o/h/L/T/h/homebrew-core ⮀ ⭠ jenv-caveat-fish  ⮀
 ⮀ la ~/.config/fish/functions/                                                                                                                           ⮂ 13:38:13 ⮂ 2021-06-02
total 32
drwxr-xr-x   6 dxia  staff   192B Jun  2 13:38 .
drwx------  16 dxia  staff   512B May 12 07:01 ..
-rw-r--r--   1 dxia  staff   816B Feb 17  2019 burk.fish
-rw-r--r--   1 dxia  staff   5.2K Sep  9  2014 fish_prompt.fish
-rw-r--r--   1 dxia  staff    55B Feb 13  2019 fish_user_key_bindings.fish
lrwxr-xr-x   1 dxia  staff    45B Apr  8 15:11 fzf_key_bindings.fish -> /opt/homebrew/opt/fzf/shell/key-bindings.fish
```

Reply:
@davidxia I think they're being installed elsewhere, according to the output:

```
fish functions have been installed to:
  /opt/homebrew/share/fish/vendor_functions.d
```

Fish should still find them there. Can you test that?

Reply:
Thanks. Didn't notice that. Is the below enough? No need for symlinks in `~/.config/fish`?

```
la /opt/homebrew/share/fish/vendor_functions.d                                                                                                         ⮂ 13:48:41 ⮂ 2021-06-02
total 0
drwxr-xr-x   4 dxia  admin   128B Jun  2 13:40 .
drwxr-xr-x  12 dxia  admin   384B Jun  2 13:40 ..
lrwxr-xr-x   1 dxia  admin    68B Jun  2 13:40 export.fish -> ../../../Cellar/jenv/0.5.4/share/fish/vendor_functions.d/export.fish
lrwxr-xr-x   1 dxia  admin    66B Jun  2 13:40 jenv.fish -> ../../../Cellar/jenv/0.5.4/share/fish/vendor_functions.d/jenv.fish
```

Reply:
jenv still works. It seems to be enough?

```
rm ~/.config/fish/functions/jenv.fish ~/.config/fish/functions/export.fish

[new terminal]

java -version                                                                                                                                          ⮂ 13:49:32 ⮂ 2021-06-02
openjdk version "11.0.10" 2021-01-19 LTS
OpenJDK Runtime Environment Corretto-11.0.10.9.1 (build 11.0.10+9-LTS)
OpenJDK 64-Bit Server VM Corretto-11.0.10.9.1 (build 11.0.10+9-LTS, mixed mode)
 dxia@dxia-mbp ⮀ ~ ⮀
 ⮀ jenv versions                                                                                                                                          ⮂ 13:49:40 ⮂ 2021-06-02
  system
* 11 (set by /Users/dxia/.java-version)
  11.0.10
  16
  corretto64-11.0.10
  openjdk64-16
 dxia@dxia-mbp ⮀ ~ ⮀
 ⮀ jenv shell 16                                                                                                                                          ⮂ 13:49:42 ⮂ 2021-06-02
 dxia@dxia-mbp ⮀ ~ ⮀
 ⮀ java -version                                                                                                                                          ⮂ 13:50:12 ⮂ 2021-06-02
openjdk version "16" 2021-03-16
OpenJDK Runtime Environment (build 16+14)
OpenJDK 64-Bit Server VM (build 16+14, mixed mode)
```

Reply:
I think so. I don't have a strong sense of what those functions do for jenv as it seems to work without them too 🤷🏻.

Reply:
Yes, the idea is that the symlinks into `~/.config/fish` should no longer be needed. Please try using `jenv` and see if it's working normally.

Reply:
Seems to be working normally for me!

Reply:
Awesome. One last thing and this should be good to go.

Reply:
Thanks. Committed your suggestions!